api/include/opentelemetry/metrics/sync_instruments.h

```diff
@@ -247,5 +247,82 @@ class UpDownCounter : public SynchronousInstrument
   }
 };
 
+#if OPENTELEMETRY_ABI_VERSION_NO >= 2
+/* A Gauge instrument that records values. */
+template <class T>
+class Gauge : public SynchronousInstrument
+{
+
+public:
+  /**
+   * Record a value
+   *
+   * @param value The measurement value. May be positive, negative or zero.
```

Comment:
not blocking the merge. Seems `T` should be integer or float types? If so, could it be asserted explicitly like below?

```
static_assert(std::is_same_v<T, int64_t> || std::is_same_v<T, double>, "Gauge<T> can only be instantiated with int64_t or double");
```